extensions/modules/file/src/main/java/org/exist/xquery/modules/file/Sync.java

```diff
@@ -55,183 +62,356 @@
-				reportError(output, "Collection not found: " + collectionPath);
-				return;
-			}
-			for (final Iterator<DocumentImpl> i = collection.iterator(context.getBroker()); i.hasNext(); ) {
-				DocumentImpl doc = i.next();
-				if (startDate == null || doc.getLastModified() > startDate.getTime()) {
-					if (doc.getResourceType() == DocumentImpl.BINARY_FILE) {
-						saveBinary(targetDir, (BinaryDocument) doc, output);
-					} else {
-						saveXML(targetDir, doc, output);
-					}
-				}
-			}
-			
-			subcollections = new ArrayList<>(collection.getChildCollectionCount(context.getBroker()));
-			for (final Iterator<XmldbURI> i = collection.collectionIterator(context.getBroker()); i.hasNext(); ) {
-				subcollections.add(i.next());
-			}
-		}
-		
-		for (final XmldbURI childURI : subcollections) {
-			final Path childDir = targetDir.resolve(childURI.lastSegment().toString());
-			saveCollection(collectionPath.append(childURI), childDir, startDate, output);
-		}
-	}
-
-	private void reportError(final MemTreeBuilder output,final String msg) {
-		output.startElement(new QName("error", FileModule.NAMESPACE_URI), null);
-		output.characters(msg);
-		output.endElement();
-	}
-
-	private void saveXML(final Path targetDir, final DocumentImpl doc, final MemTreeBuilder output) {
-		Path targetFile = targetDir.resolve(doc.getFileURI().toASCIIString());
-		final SAXSerializer sax = (SAXSerializer)SerializerPool.getInstance().borrowObject( SAXSerializer.class );
-		try {
-			if (Files.exists(targetFile) && Files.getLastModifiedTime(targetFile).compareTo(FileTime.fromMillis(doc.getLastModified())) >= 0) {
-				return;
-			}
-    	    boolean isRepoXML = Files.exists(targetFile) && FileUtils.fileName(targetFile).equals("repo.xml");
-
-			output.startElement(new QName("update", FileModule.NAMESPACE_URI), null);
-			output.addAttribute(new QName("file", XMLConstants.NULL_NS_URI), targetFile.toAbsolutePath().toString());
-			output.addAttribute(new QName("name", XMLConstants.NULL_NS_URI), doc.getFileURI().toString());
-			output.addAttribute(new QName("collection", XMLConstants.NULL_NS_URI), doc.getCollection().getURI().toString());
-			output.addAttribute(new QName("type", XMLConstants.NULL_NS_URI), "xml");
-			output.addAttribute(new QName("modified", XMLConstants.NULL_NS_URI), new DateTimeValue(new Date(doc.getLastModified())).getStringValue());
+    }
+
+    public Sync(final XQueryContext context, final FunctionSignature signature) {
+        super(context, signature);
+    }
+
+    @Override
+    public Sequence eval(final Sequence[] args, final Sequence contextSequence) throws XPathException {
+
+        if (!context.getSubject().hasDbaRole()) {
+            throw new XPathException(this, "Function file:sync is only available to the DBA role");
+        }
+
+        final String collectionPath = args[0].getStringValue();
+        final String target = args[1].getStringValue();
+        final HashMap<String, Sequence> options = getOptions(args[2]);
```

Comment:
As recognised good practice, the LHS (Left Hand Side) of a variable assignment in Java should always use the lowest-common Interface rather than the Implementation of that Interface. That is to say that this should likely be:
```java
final Map<String, Sequence> options = getOptions(args[2]);
```